projects/vdk-core/src/vdk/internal/builtin_plugins/run/cli_run.py

```diff
@@ -66,6 +71,45 @@ def __split_into_chunks(exec_steps: List, chunks: int) -> List:
                 + (quotient + 1 if i < remainder else quotient)
             ]
 
+    @staticmethod
+    def __warn_on_python_version_disparity(
+        context: CoreContext, job_directory: pathlib.Path
+    ):
+        log_config_type = context.configuration.get_value(vdk_config.LOG_CONFIG)
+        if log_config_type == "LOCAL":
+            # Get the local python installation's version.
+            python_env_version = sys.version_info
+            local_py_version = f"{python_env_version.major}.{python_env_version.minor}"
+
+            # Get the python_version set in the config.ini if any.
+            job_path = job_directory.resolve()
+            try:
+                config = JobConfig(data_job_path=job_path)
+            except VdkConfigurationError as e:
+                log.info(
+                    f"An exception occurred while loading job configuration. Error was {e}"
+                )
+                return
+            configured_python_version = config.get_python_version()
+
+            if not configured_python_version:
+                return
+
+            pattern = r"^\d+\.\d+"
+            version_match = re.match(pattern, configured_python_version)
+
+            if version_match:
+                extracted_configured_version = version_match.group()
+                if extracted_configured_version != local_py_version:
+                    log.warning(
+                        f"""
+                        {os.linesep + (' ' * 18) + ('*' * 80)}
```

Comment:
Let's improve the message. 

As a reminder we have a recommendation on what good message should contain: https://github.com/vmware/versatile-data-kit/wiki/Coding-Standard#error-handling

Reply:
Hi, @tozka 

I've updated the warning message, following the error-handling standard as much as possible. Thanks